api/preprints/serializers.py

```diff
@@ -0,0 +1,147 @@
+        'date_modified',
+        'contributors',
+        'provider',
+        'subjects',
+        'doi'
+    ])
+
+    title = ser.CharField(required=False)
+    subjects = JSONAPIListField(required=False, source='get_preprint_subjects')
+    provider = ser.CharField(source='preprint_provider', required=False)
+    date_created = ser.DateTimeField(read_only=True, source='preprint_created')
+    date_modified = ser.DateTimeField(read_only=True)
+    id = IDField(source='_id', required=False)
+    abstract = ser.CharField(source='description', required=False)
+    tags = JSONAPIListField(child=NodeTagField(), required=False)
+    doi = ser.CharField(source='preprint_doi', required=False)
+
+    primary_file = PrimaryFileRelationshipField(
+        related_view='files:file-detail',
+        related_view_kwargs={'file_id': '<preprint_file._id>'},
+        lookup_url_kwarg='file_id',
+        read_only=False
+    )
+
+    files = RelationshipField(
+        related_view='nodes:node-providers',
+        related_view_kwargs={'node_id': '<pk>'}
+    )
+
+    links = LinksField(
+        {
+            'self': 'get_preprint_url',
+            'html': 'get_absolute_html_url',
+            'doi': 'get_doi_url'
+        }
+    )
+
+    contributors = RelationshipField(
+        related_view='preprints:preprint-contributors',
+        related_view_kwargs={'node_id': '<pk>'},
+        related_meta={'count': 'get_contrib_count'},
+    )
+
+    class Meta:
+        type_ = 'preprints'
+
+    def get_preprint_url(self, obj):
+        return absolute_reverse('preprints:preprint-detail', kwargs={'node_id': obj._id})
+
+    def get_absolute_url(self, obj):
+        return self.get_preprint_url(obj)
+
+    def get_doi_url(self, obj):
+        return 'https://dx.doi.org/{}'.format(obj.preprint_doi)
+
+    def create(self, validated_data):
+        node = validated_data.pop('node')
+        if node.is_preprint:
+            raise Conflict('This node already stored as a preprint, use the update method instead.')
+        auth = get_user_auth(self.context['request'])
+
+        primary_file = validated_data.pop('primary_file', None)
+        if not primary_file:
+            raise exceptions.ValidationError(detail='A primary file is required to create a preprint.')
+
+        preprint_subjects = validated_data.get('preprint_subjects', None)
+        if not preprint_subjects:
+            raise exceptions.ValidationError(detail='Subjects are required to create a preprint.')
+
+        try:
+            node.set_preprint_file(primary_file, auth, save=False)
+        except PermissionsError:
+            raise exceptions.PermissionDenied('Not authorized to create a preprint from this node.')
+        except ValueError as e:
+            raise exceptions.ValidationError(detail=e.message)
+
+        if node._id != validated_data.pop('_id'):
+            raise exceptions.ValidationError('The node id in the URL does not match the id in the request body.')
```

Comment:
This is a red flag that maybe we shouldn't be architecting these endpoints this way. Typically, `List` endpoints accept `POST`s to create, not `Detail` endpoints, so there shouldn't be a node id in the url for a call to `.create`